File: java/core/src/core/connector/sync/StoreConnector.java
/**
 * Author: Timothy Prepscius
 * License: GPLv3 Affero + keep my name in the code!
 */

package core.connector.sync;

import java.util.List;

import core.connector.ConnectorException;
import core.connector.FileInfo;

public interface StoreConnector 
{	
	public void open () throws ConnectorException;
	public void close () throws ConnectorException;
	
	List<FileInfo> listDirectory (String path) throws ConnectorException;
	void createDirectory (String path) throws ConnectorException;
	
	byte[] get (String path) throws ConnectorException;
	byte[] get (String path, long size) throws ConnectorException;

	void put (String path, byte[] contents) throws ConnectorException;
	void move (String from, String to) throws ConnectorException;
	void delete (String path) throws ConnectorException;
	
	public boolean ensureDirectories (String ... folders);
}

File: cpp/core/mailiverse/core/connector/StoreConnector.cpp
/**
 * Author: Timothy Prepscius
 * License: BSD + keep my name in the code!
 */

#include "StoreConnector.h"
#include "mailiverse/utilities/Log.h"

using namespace mailiverse::core::connector;

StoreConnector::StoreConnector ()
{
	LogDebug(mailiverse::core::connector, "StoreConnector");
}

StoreConnector::~StoreConnector ()
{
	LogDebug(mailiverse::core::connector, "~StoreConnector");
}

StoreConnector::FileListing StoreConnector::list (const std::string &path, const std::pair<FileInfo::Date,FileInfo::Date> &dateRange) throws_ (ConnectorException)
{
	FileListing results;
	FileListing files = list(path);
	for (auto &i : files)
	{
		if ((dateRange.first < i.date || dateRange.first.getTime()==-1) &&
			(i.date < dateRange.second || dateRange.second.getTime()==-1))
		{
			results.push_back(i);
		}
	}

	return results;
}
